lib/introspection.js

```diff
@@ -71,28 +69,120 @@ const parseSignature = (
   return signature;
 };
 
-const introspect = (
-  // Introspect interface
-  namespace // hash of interfaces
-  // Returns: hash of hash of record, { method, title, parameters }
-) => {
+const parseFunction = (fn, text) => {
+  const fnData = fn + '';
+  const fnIndex = text.indexOf(fnData);
+  const lines = text.slice(0, fnIndex)
+    .split('\n')
+    .map(line => line.trim());
+  const comments = [];
+
+  for (let i = lines.length - 2; i >= 0; i--) {
+    if (lines[i].indexOf('// ') !== 0) break;
+    comments.unshift(lines[i].slice(3));
+  }
+  return comments;
+};
+
+const parseComments = (lines) => {
```

Comment:
We need description and use examples for new implementation, also: change `(lines) => {` to `lines => {`